### ISSUE TITLE
Mobile first approach theme

### ISSUE THREAD

**Issue by domaingood** [backdrop/backdrop-issues#4131]
**Drupal Mobile First Design Strategy for Websites**

**Proposed solution**
https://www.drupal.org/project/boilerplate

**Additional information**
Based on the excellent HTML5 Boilerplate template, this theme was built to be a lean responsive design base theme with heavy SASS support. Inspired by the Basic, Boron, Zen, Panopoly, and Sassy themes.

Features
Responsive Panel set based on the great Panopoly distribution. (new in 1.8)
Custom SASS mixins for breakpoints and REM sizing. (new in 1.8)
Drush integration. As simple as "drush boilerplate new-theme-name". You will also be asked if you are using SASS or not and the links to the compiled css will be automatically changed in the .info file. (new in 1.8)
Modernizr.js allows for HTML5 element use in older browsers and browser feature detection classes included in HTML tag.
Box-sizing: border-box layout with IE shim - CSS padding and border width no longer included in width calculations. Easily removed if not wanted.
SASS and Compass support with SCSS syntax (Compass is required for the default installation but can be removed if not wanted)
Three default responsive layouts at <480 (iPhone), 480-800 (netbooks/tablets), and >800 (PCs)
jQuery.mobileDropDown to automatically change navigation to drop down menus on mobile devices
Zen wireframing
Zen Breadcrumb
Console.log() protection from IE errors
If you don't want the SASS support, you'll need to change your CSS links in the .info file, but the drush command will ask and change it accordingly.

**Comment by klonos** [comment]
The possibility of SASS or SCSS has been brought up in the past, but it has been dismissed from being offered with a Backdrop core theme. Reasons being that we do not want to introduce the same level of complexity, nor higher bar of entry for novice developers and code tinkerers. Referring to [our principles](https://backdropcms.org/philosophy):

> 2. **Simplicity: Write code for the majority**. Backdrop will aim to be easy to learn and build upon, even for those with a minimal amount of technical knowledge. Direct implementations will be chosen over abstraction, and how things work should be immediately clear and easily documentable. While the more vocal parts of our community may be more experienced developers, we always need to keep in mind the developers who are newer to the project.

**Comment by domaingood** [comment]
Thank you for explain Simplicity @klonos But SASS or SCSS most popular now a days.We can give front end developer option CSS & SASS or SCSS. So They can use what their need and write code. End of day we are creating CSS :).

**Comment by klonos** [comment]
I'm afraid that you are thinking as an experienced developer on this one @domaingood.

>End of day we are creating CSS :).

CSS is creating CSS; any other option like SASS/SCSS, you are requiring people to install things that **compile** CSS for them. This makes the bar of entry higher than our intended audience.

One of the main reasons why Backdrop was forked was because Drupal decided to introduce with what is "popular", like twig, composer, symfony etc. If that is what a developer needs to be using, perhaps D8 is a better option in that case.

**Comment by stpaultim** [comment]
I believe that we do want more options for developers in contrib. We do need more contrib themes that are built around these principles and include SASS option. Can we start there, by improving the options in contrib?

**Comment by domaingood** [comment]
@stpaultim Thanks for understand. Like [boilerplate](https://www.drupal.org/project/boilerplate) not in core Also Bootstrap but both are popular.I just want to ported to backdrop contrib.

**Comment by klonos** [comment]
Yes contrib is the place to go if a theme requires anything other than CSS. @domaingood you may want to apply to be a contrib maintainer here, in order to get started: https://github.com/backdrop-ops/contrib/issues

Please have a read trough https://github.com/backdrop-ops/contrib/blob/master/README.md to understand the process.

**Comment by stpaultim** [comment]
@domaingood - I'd be interested in collaborating on porting a theme like boilerplate to contrib. I'll set up a github repo where we can work on this.

**Comment by domaingood** [comment]
@stpaultim - I'd be interested too in collaborating on porting a theme like boilerplate to contrib.

**Comment by stpaultim** [comment]
@domaingood 

1. I have created a backdropcms project for Boilerplate: https://github.com/backdrop-contrib/boilerplate
2. If you apply to the contrib group as @klonos suggested, I am happy to make you a co-maintainer of this project - or turn it over to you completely. 
3. In the meantime, feel free to create issues in the issue queue for that project and/or submit PR

Note, I have not yet even installed this site on a Drupal 7 site. So, I'm not sure what we are getting into or what is involved in this port. I look forward to learning a lot and collaborating with you.
